abstract_specification.md

```diff
@@ -685,14 +695,30 @@ The exact contents of each platform’s entry in the configuration mapping are f
 - In situations where a configuration setting of the same name, but with different type, semantics, or values will be used across operations, a separate string key should be provided for that setting for each operation. This allows for the same `platform_config` data to be reused across multiple operations by the user.
   - For example, a storage backend may provide a way to read and write data in a given block size. However, the performance characteristics of these operations may be very different. The implementation should provide `read_block_size` and `write_block_size` parameters (or use some similar disambiguation strategy) rather than only allowing a single `shard_size` parameter.
 
-### Implementation and usage guidelines
+#### Implementation and usage guidelines
 
 The configuration passed in the `platform_config` object is intended for operation-specific tuning (though it may make sense for user code to use the same `platform_config` across multiple operations). A `platform_config` should only be used for the operation it was provided to (and to directly dependent operations); it should not be stored for later calls. Environmental configuration (like login credentials or backend storage region) should be provided via the (to-be-defined) Context object.
 
 Operations should not *require* a `platform_config` entry to complete; the platform should use a sensible default if a given configuration value is not provided. Required environmental data generally belongs in the Context object.
 
 A platform should only ever examine its own entry in the `platform_config` mapping. Since the structure and contents of each entry is wholly implementation-defined, one platform cannot make any assumptions at all about another’s configuration, and for predictability should avoid even trying to extract any information from any other configuration entries.
 
+### Long-lived context data
```

Comment:
A point of confusion:   as I read this, the `context` field is read-only, has no user-accessible operations, and is implementation specific.  So why specify it - why not treat it as hidden state?  How does the user "set" the context?

I think the confusion is that the user _can_ specify a context in some operations (eg, `open`).  So perhaps change the first sentence to:  "... described above, the user can specify a `context` value in certain operations.  The context value is an implementation-dependent, multi-operation configuration object, which will provide session-like state."

Do we also need to specify how it is inherited/passed, and the lifespan?  The text says "global state", which seems to imply I can only have one context?  But I suspect that there are operations where I can "set" the context, and others that will inherit it.

Lastly: do we need the concept of a default context?  I don't think we do, but it might simplify the description of how the context propagates.

Reply:
Are these changes universal, to all implementations?  Or are they Python ABI specific?   If only the latter (which seems quite possible), then they probably don't belong in this spec.

Reply:
with the example, I think this is sufficiently clear (for now).